File: src/logic/data-structures/frame.cpp
#include "frame.hpp"

namespace logic
{
	namespace data_structures
	{
		frame::frame()
		{
			handle = nullptr;
			size = 0;
		}

		frame::frame(rs2::frame& f, int width, int height)
			: width(width), height(height)
		{
			size = f.get_data_size();
			handle = new char[size];
			memcpy(handle, f.get_data(), size);
		}

		void* frame::get_data()
		{
			return handle;
		}

		void frame::free()
		{
			if (handle != nullptr)
				delete[] handle;
			
			handle = nullptr;
		}
	}
}

File: wrappers/csharp/HoloArch.HoloScan/Hal/Frame.cs
namespace HoloArch.HoloScan
{
    using System;
    using System.Runtime.InteropServices;

	public class Frame
	{
		protected IntPtr handle;
		protected int size;
		
		public Frame(IntPtr handle, int size, int height, int width)
        {
			this.handle = handle;
			this.size = size;
			Width = width;
			Height = height;
        }

		public int Width { get; private set; }
		
		public int Height { get; private set; }

		IntPtr getData()
        {
			return handle;
        }

		void free()
        {

        }
	}
}